File: Arrays&Strings/removeDuplicates.cpp
// 26) Remove Duplicates from Sorted Array

#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        nums.erase(unique(nums.begin(), nums.end()), nums.end()); 

        return nums.size(); 
    }
};

File: Linked_List/removeDuplicates.py
# REmove duplicates from sorted list II

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def deleteDuplicates(self, head):
        if not head or not head.next: return head

        right = ListNode(0)
        right.next = head
        left = right
        
        while head:
            while head.next and head.val == head.next.val:
                head = head.next

            if left.next == head:
                left = left.next
            else:
                left.next = head.next # remove nodes from duplicates

            head = head.next

        return right.next



        